### PR TITLE
docs(srv): add SEM x Jobs documentation MTA-5199

### PR DESCRIPTION
**Pull request by SamyOubouaziz**
### Your checklist for this pull request

- [ ] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description

Please describe what you added or changed.

### CODE REVIEW

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+You can reference any secret stored in Secret Manager in a job, without having to hardcode any sensible data.
+
+<Macro id="requirements" />
+
+- A Scaleway account logged into the [console](https://console.scaleway.com)
+- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
+- [Created a Serverless Job](/serverless/jobs/how-to/create-job-from-scaleway-registry/)
+- [Created a secret](/identity-and-access-management/secret-manager/how-to/create-secret/)
+
+## Add a secret to a job
```
Comment (thomas-tacquet) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:28]:
```suggestion
## Reference a secret to a job
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your container. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your container.
+
+  - **Environment variable**: passes the encrypted value of your secret to your job as a variable. This method is recommended for small information, such as passwords, or API secret keys. For example, if you name this variable `MY_SECRET`, calling `$MY_SECRET` in your container will return the value of the selected secret in a secure way.
+
+6. Click **Add reference** to add the secret to your Serverless Job. Optionally, tick the **Add another reference** to add a new secret right away, then repeat steps 4 to 6.
+
+The secret is now referenced in your Serverless Job, and can be used within the container.
+
+## Update a secret in a job
```
Comment (thomas-tacquet) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:48]:
```suggestion
## Update a secret reference from a job
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  validation: 2024-10-27
+  posted: 2024-10-27
+categories:
+  - serverless
+  - jobs
+---
+
+Serverless Jobs seamlessly integrates with [Secret Manager](/identity-and-access-management/secret-manager/), which allows you to store, manage, and access sensitive information, such as credentials, SSH keys, SSL/TLS certificates, or any key/value pairs you need to secure.
+
+You can reference any secret stored in Secret Manager in a job, without having to hardcode any sensible data.
```
Comment (thomas-tacquet) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:19]: I think you can add somewhere in the doc that "each access to a Secret by a Job Run generates a call to the Secret Manager API, refer to [Secret Manager pricing](https://www.scaleway.com/en/pricing/security-and-account/) for more information"

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Click **Jobs** in the **Serverless** section of the side menu. The jobs page displays.
+
+2. Click the name of the job for which you want to update a secret, then open the **Settings** tab.
+
+3. In the **Secret references** section, click the <Icon name="edit" /> icon next to the secret reference you want to update. A pop-up displays.
+
+4. Update the secret version if needed, then click **Update** to save your changes, or click **Select reference method** to continue.
+
+5. Either update the location of the file, or the name of the environment, variable then click **Update reference** to confirm your changes.
```
Comment (nerda-codes) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:58]:
```suggestion
5. Either update the location of the file, or the name of the environment variable, then click **Update reference** to confirm your changes.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+<Message type="note">
+You cannot change the reference method of an existing secret. To do so, delete the secret reference within the job, then create it again with the desired reference method.
```
Comment (nerda-codes) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:61]:
```suggestion
You cannot change the reference method of an existing secret. You have to delete the secret reference within the job first, then create it again with the desired reference method.
```

File: serverless/jobs/quickstart.mdx [hunk 6]
```diff
@@ -62,10 +62,12 @@ This page explains how to create a job definition with the latest Alpine Linux i
 
 3. From the **Overview** tab, click **Run job**.
 
-The execution appears in the **Job runs** section of the **Overview** tab.
+  The execution appears in the **Job runs** section of the **Overview** tab.
+
+4. Click <Icon name="more"> next to the last execution in the **Job runs** section, then click **Logs** to access your job's logs.
```
Comment (nerda-codes) [inline serverless/jobs/quickstart.mdx:67]:
```suggestion
4. Click the <Icon name="more"> icon next to the last execution in the **Job runs** section, then click **Logs** to access your job's logs.
```

File: serverless/jobs/quickstart.mdx [hunk 7]
```diff
@@ -62,10 +62,12 @@ This page explains how to create a job definition with the latest Alpine Linux i
...
 
 <Message type="note">
-  Refer to [How to monitor a job](/serverless/jobs/how-to/monitor-job/) to see the logs of the job you just executed.
+  Make sure that you [retrieved your Grafana credentials](/observability/cockpit/how-to/retrieve-grafana-credentials/) before accessing your job's logs.
```
Comment (nerda-codes) [inline serverless/jobs/quickstart.mdx:70]:
```suggestion
  Make sure that you [have retrieved your Grafana credentials](/observability/cockpit/how-to/retrieve-grafana-credentials/) before accessing your job's logs.
```

File: faq/serverless-jobs.mdx [hunk 8]
```diff
@@ -119,4 +119,8 @@ skopeo copy --override-os linux docker://docker.io/alpine:latest docker://rg.fr-
 
 Scaleway Serverless Jobs does not currently support Scaleway VPC or Private Networks, though this feature is under development.
 
-To add network restrictions on your resource, consult the [list of prefixes used at Scaleway](https://www.scaleway.com/en/peering/). Serverless resources do not have dedicated or predictable IP addresses.
\ No newline at end of file
+To add network restrictions on your resource, consult the [list of prefixes used at Scaleway](https://www.scaleway.com/en/peering/). Serverless resources do not have dedicated or predictable IP addresses.
+
+## Can I securely use sensitive information with Serverless Jobs?
+
+Yes, you can use sensitive data such as API secret keys, passwords, TLS/SSL certificates, or tokens. Serverless Jobs seamlessly integrates with [Secret Manager](/identity-and-access-management/secret-manager/), which allows you to securely reference sensible information within your jobs. Refer to the [dedicated documentation](/serverless/jobs/how-to/reference-secret-in-job/) for more information.
```
Comment (RoRoJ) [inline faq/serverless-jobs.mdx:126]:
```suggestion
Yes, you can use sensitive data such as API secret keys, passwords, TLS/SSL certificates, or tokens. Serverless Jobs seamlessly integrates with [Secret Manager](/identity-and-access-management/secret-manager/), which allows you to securely reference sensitive information within your jobs. Refer to the [dedicated documentation](/serverless/jobs/how-to/reference-secret-in-job/) for more information.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx (hunk 3 above)
Comment (RoRoJ) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:19]:
```suggestion
You can reference any secret stored in Secret Manager in a job, without having to hardcode any sensitive data.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx (hunk 1 above)
Comment (RoRoJ) [reply serverless/jobs/how-to/reference-secret-in-job.mdx:28]: or "in a job", if the verb is being changed to Reference

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+3. In the **Secrets references** section, click **+ Add secret reference**. A pop-up displays.
+
+4. Select the secret you want to reference, and the desired version, then click **Select reference method**.
+
+5. Select the desired reference method:
+
+  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your container. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your container.
+
+  - **Environment variable**: passes the encrypted value of your secret to your job as a variable. This method is recommended for small information, such as passwords, or API secret keys. For example, if you name this variable `MY_SECRET`, calling `$MY_SECRET` in your container will return the value of the selected secret in a secure way.
```
Comment (RoRoJ) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:42]:
```suggestion
  - **Environment variable**: passes the encrypted value of your secret to your job as a variable. This method is recommended for small pieces of information, such as passwords, or API secret keys. For example, if you name this variable `MY_SECRET`, calling `$MY_SECRET` in your container will return the value of the selected secret in a secure way.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx (hunk 4 above)
Comment (RoRoJ) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:58]:
```suggestion
5. Either update the location of the file, or the name of the environment variable, then click **Update reference** to confirm your changes.
```

**Review by bene2k1 — APPROVED** [review]

**Review by nerda-codes — APPROVED** [review]

File: serverless/jobs/how-to/create-job-from-external-registry.mdx [hunk 10]
```diff
@@ -42,6 +42,7 @@ Private external container registries are currently not supported.
         <Message type="note">
         Encode your environment variables to `base64` if they are too large, and contain carriage returns.
         </Message>
+    - Add the desired [secret references](/serverless/jobs/how-to/reference-secret-in-job/) to your job
```
Comment (jcirinosclwy) [inline serverless/jobs/how-to/create-job-from-external-registry.mdx:45]:
```suggestion
    - Add the desired [secret references](/serverless/jobs/how-to/reference-secret-in-job/) to your job.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 11]
```diff
@@ -0,0 +1,80 @@
+---
+meta:
+  title: How to reference secrets in Serverless Jobs
+  description: Steps to reference secrets from Secret Manager in your Serverless Jobs.
+content:
+  h1:  How to reference secrets in Serverless Jobs
+  paragraph: Steps to reference secrets from Secret Manager in your Serverless Jobs.
+tags: serverless jobs secrets secret-manager reference environment variable
```
Comment (jcirinosclwy) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:8]:
```suggestion
tags: serverless jobs secrets secret-manager environment-variable
```

**Review by jcirinosclwy — APPROVED** [review]

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+2. Click the name of the job to which you want to add a secret, then open the **Settings** tab.
+
+3. In the **Secrets references** section, click **+ Add secret reference**. A pop-up displays.
+
+4. Select the secret you want to reference, and the desired version, then click **Select reference method**.
+
+5. Select the desired reference method:
+
+  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your container. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your container.
```
Comment (RamyChaabane) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:42]:
```suggestion
  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your job. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your job
```

I think, from a product perspective, the client doesn’t need to be aware of the underlying technologies we use.

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+  - **Environment variable**: passes the encrypted value of your secret to your job as a variable. This method is recommended for small pieces of information, such as passwords, or API secret keys. For example, if you name this variable `MY_SECRET`, calling `$MY_SECRET` in your container will return the value of the selected secret in a secure way.
```
Comment (RamyChaabane) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:44]:
```suggestion
  - **Environment variable**: passes the encrypted value of your secret to your job as a variable. This method is recommended for small pieces of information, such as passwords, or API secret keys. For example, if you name this variable `MY_SECRET`, calling `$MY_SECRET` in your job will return the value of the selected secret in a secure way.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+6. Click **Add reference** to add the secret to your Serverless Job. Optionally, tick the **Add another reference** to add a new secret right away, then repeat steps 4 to 6.
+
+The secret is now referenced in your Serverless Job, and can be used within the container.
```
Comment (RamyChaabane) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:48]:
```suggestion
The secret is now referenced in your Serverless Job, and can be used within the job.
```

File: serverless/jobs/how-to/reference-secret-in-job.mdx [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+2. Click the name of the job for which you want to update a secret, then open the **Settings** tab.
+
+3. In the **Secret references** section, click the <Icon name="edit" /> icon next to the secret reference you want to update. A pop-up displays.
+
+4. Update the secret version if needed, then click **Update** to save your changes, or click **Select reference method** to continue.
+
+5. Either update the location of the file, or the name of the environment variable, then click **Update reference** to confirm your changes.
+
+<Message type="note">
+You cannot change the reference method of an existing secret. You have to delete the secret reference within the job first, then create it again with the desired reference method.
```
Comment (RamyChaabane) [inline serverless/jobs/how-to/reference-secret-in-job.mdx:63]:
```suggestion
You cannot change the reference method of an existing secret. To modify it, first delete the current secret reference within the job, then create a new one with the desired reference method.
```

**Pull request merged**